common/planners/src/plan_node.rs

```diff
@@ -127,4 +134,27 @@ impl PlanNode {
     pub fn input(&self, n: usize) -> Arc<PlanNode> {
         self.inputs()[n].clone()
     }
+
+    pub fn set_inputs(&mut self, inputs: Vec<&PlanNode>) -> Result<()> {
+        if inputs.is_empty() {
+            return Result::Err(ErrorCodes::BadPlanInputs("Inputs must not be empty"));
+        }
+
+        match self {
+            PlanNode::Stage(v) => v.set_input(inputs[0]),
+            PlanNode::Projection(v) => v.set_input(inputs[0]),
+            PlanNode::Expression(v) => v.set_input(inputs[0]),
+            PlanNode::AggregatorPartial(v) => v.set_input(inputs[0]),
+            PlanNode::AggregatorFinal(v) => v.set_input(inputs[0]),
+            PlanNode::Filter(v) => v.set_input(inputs[0]),
+            PlanNode::Having(v) => v.set_input(inputs[0]),
+            PlanNode::Limit(v) => v.set_input(inputs[0]),
+            PlanNode::Explain(v) => v.set_input(inputs[0]),
+            PlanNode::Select(v) => v.set_input(inputs[0]),
+            PlanNode::Sort(v) => v.set_input(inputs[0]),
+            _ => {}
```

Comment:
Maybe better to return an error for the other types of PlanNode; otherwise, the error can be propagated to somewhere else.